### ISSUE TITLE
Remove `graphql.rebuild_schema()` from public API

### ISSUE THREAD

**Issue by olirice** [supabase/pg_graphql#144]
Instead of rebuilding in a ddl event listener, increment a sequence tracking the schemas version.

Check the current value of the sequence inside the `resolve` function and, if it differs from the cached schema version, call `graphql.rebuild_schema` before executing the query.

**Comment by prescience-data** [comment]
May be relevant: 

On a newly created project, `pg_graphql` seems to be enabled by default, however it only creates the `graphql_public` schema.

When toggling the `pg_graphql` extension off for the first time, an error message is presented saying that the extension could not be found.

Works as expected once the extension is toggled back on.

**Comment by olirice** [comment]
Thank you

This issue is to remove the need for users to call rebuild_schema. It's an enhancement goal rather than a bug

**Comment by prescience-data** [comment]
So is the expected behaviour that the "old" / original schema returns after toggling off than on?

**Comment by olirice** [comment]
You should not need to toggle the extension off or on

As of last week, pg_graphql is enabled on new projects by default and you can call `graphql.rebuild_schema()` to rebuild the schema as needed (the permissions issue you previously faced was fixed)

**Comment by prescience-data** [comment]
There's a chance I created the project in some interim state, but perhaps have a look at the behaviour of a fresh project on live Supabase platform and inspect the live database with a direct postgres connection to see the issue above.

The UI "extension not found" error popup is triggered only once when disabling the extension for the first time, but still caused an error.

**Comment by olirice** [comment]
For the first few days of pg_graphql general availability on the supabaes platform the extension was disabled by default. Now it is enabled by default and you can see the `public_graphql` + `graphql` schemas immediately after creating a project. 

I just confirmed on a new project, but if you see something else, please let me know

**Comment by prescience-data** [comment]
@olirice  
When running migrations as `postgres` user on a fresh project (just created minutes ago) I am still getting:

```

[2022-04-23 20:22:17] 0 row(s) affected in 100 ms
SELECT graphql.rebuild_schema( )
[2022-04-23 20:22:17] [3F000] ERROR: schema "graphql" does not exist
[2022-04-23 20:22:17] Position: 8
[2022-04-23 20:22:17] Undetect Dev: 2 of 1 statements executed, 1 failed in 210 ms
```

**Comment by prescience-data** [comment]
Additionally, this is the UI error presented when turning off the `pg_graphql` extension for the first time (does not occur after the first time, meaning there is some issue with the setup triggered by new project creation): 

![image](https://user-images.githubusercontent.com/65471523/164890646-8bd396da-8f00-44f3-8436-a895cab43b68.png)

**Comment by olirice** [comment]
> When running migrations as postgres user on a fresh project (just created minutes ago) I am still getting:
> 
> 
> [2022-04-23 20:22:17] 0 row(s) affected in 100 ms
> SELECT graphql.rebuild_schema( )
> [2022-04-23 20:22:17] [3F000] ERROR: schema "graphql" does not exist
> [2022-04-23 20:22:17] Position: 8
> [2022-04-23 20:22:17] Undetect Dev: 2 of 1 statements executed, 1 failed in 210 ms
> After toggling the pg_graphql extension off then back on, it behaves normally.

this looks like you're running locally via the CLI. Is that correct?

**Comment by prescience-data** [comment]
Migrations are via IDE connected to hosted Supabase instance.

The error is via the supabase.com GUI if that helps?

**Comment by olirice** [comment]
I can not reproduce that error. As far as I can see, new projects have pg_graphql created by default, the graphql schema exists, and the default graphql request is served without issue

If you're having trouble with initial project setup, please create a new issue so we can keep this issue single purpose.
In that issue please attempt to reproduce the state you're seeing in a new project and send me the project ref so I can investigate

thanks

**Comment by joesaunderson** [comment]
Not sure if this is related - but I think since this PR dropped, I can no longer access GraphQL requests in Supbase outside of the `.resolve()` method in SQL (i.e via an IDE like Postman or Altair). 

All I get when I run an introspection query is `heartbeat` - whereas I can query multiple entities using `resolve()`...  any ideas>

**Comment by joesaunderson** [comment]
Some debugging:

Via SQL:

```
select graphql.resolve($$
    {
      usersCollection(first: 1) {
        edges {
          node {
            id
            email
          }
        }
      }
    }
$$);
```

```
{"data":{"usersCollection":{"edges":[{"node":{"id":"e2fdd949-11d5-4325-9c6f-4b6071f779a5","email":"xxxx@gmail.com"}}]}}}
```

VIA API:

```
{
  usersCollection(first: 1) {
    edges {
      node {
        id
        email
      }
    }
  }
}
```

```
{
  "data": null,
  "errors": [
    {
      "message": "Unknown field 'usersCollection' on type 'Query'"
    }
  ]
}
```

**Comment by olirice** [comment]
@joesaunderson could you please send your project reference? That does sound like a schema rebuilding issue
